esp-hal/src/soc/esp32s3/cpu_control.rs

```diff
@@ -259,7 +259,7 @@ impl<'d> CpuControl<'d> {
     where
         F: FnOnce(),
     {
-        match START_CORE1_FUNCTION.take() {
+        match &mut (*&raw mut START_CORE1_FUNCTION).take() {
```

Comment:
This is raising the WTF/minute metric by way too much for me.

Reply:
Just a jump scare to keep you on your toes :D, I'll change it to `ptr_addr!` instead

Reply:
I think the syntax is somewhat less of an issue, the taking a reference of a dereferenced raw pointer is just... weird

Reply:
We should maybeport this code to use AtomicPtr?

Reply:
Probably, but I'm not going to do that here, I want to release the compiler today so just `allow(static_mut_refs)` for now